common/src/command.rs

```diff
@@ -17,13 +95,21 @@ impl<T: CommandType> CommandTypeDynamic for T {
 
 #[derive(Debug, Clone)]
 pub struct Speak(pub String);
+
+#[derive(Debug, Clone)]
+pub struct SpeechPriority(pub Priority);
+
 impl CommandType for Speak {
```

Comment:
why is this mechanism so convoluted? what are you trying to do with that extra trait, discriminants and so on? what's wrong with the design discussed yesterday in matrix? all this complication isn't needed, imo, though maybe there's something I'm missing. The same way I'm asking these questions here, someone new to the codebase might do the same, so where intent is clear but I'm missing it, code documentation is required

Reply:
It is so we can distinguish between discriminants in the `Command` enum. This is a similar system to what is used in `atspi` with the `BusProperties` and `EventTypeProperties` traits that distinguishes `Command` enums based on its internal discriminants.

Reply:
well yeah, discussed in the previous comments, handlers shouldn't do it that way, but anyway closing the thread as resolved